public/compute/first_contact.md

```diff
@@ -24,41 +86,84 @@ In Molgenis Compute, anything can be a parameter.
+In this workflow, the parameter values are listed in two files:
 
-You can generate a template for a new workflow using the command:
+### parameters.csv
+|input|
+|-----|
+|hello|
+|bye  |
 
-```bash
-  sh molgenis_compute.sh --create myfirst_workflow
+The `input` parameter has two different values, hello and bye.
+In `step1`, the `input` parameter gets mapped to the `in` parameter of `protocols/step1.sh`.
+The parameter file lists two values for this parameter so this script will be created twice, one version for `input=hello`, and one for `input=bye`
+
+### workflow.defaults.csv
+|workflowName|creationDate|
+|------------|------------|
+|myFirstWorkflow|today    |
+
+These parameter values will be inserted into the template for `step2`.
+Let's take a look at `protocols/step2.sh`
+
+```FreeMarker
+#string wf
+#string date
+#list strings
+
+echo "Workflow name: ${wf}"
+echo "Created: ${date}"
+
+echo "Result of step1.sh:"
+for s in "${strings[@]}"
+do
+    echo ${s}
+done
+
+
+echo "(FOR TESTING PURPOSES: your runid is ${runid})"
 ```
 
-This will create a new directory for the workflow:
+The first three lines are the header listing the input parameters of the protocol.
+In `step2` maps the parameter value `workflowName=myFirstWorkflow` from file `workflow.defaults.csv` to protocol input `wf`. So when the script for `step2` gets generated, the value `myFirstWorkflow` will be filled in where it says `${wf}`.
+
+The output from `step1.sh` is parameter value `step1.out`, and `step2` maps it to parameter `strings` of this protocol.
+
+But `step1.sh` will be run twice! So `step1.out` will have two different values, namely `hello_hasBeenInStep1` and `bye_hasBeenInStep1`. Of course, we could run `step2` twice, once for each value of `step1.out`. But the protocol uses `#list` to indicate that it can process multiple values for this input parameter in one single go. So a single instance of `step2.sh` will be generated. The `strings` input parameter will have a list value `strings=hello_hasBeenInStep1,bye_hasBeenInStep1`.
+
+## Generate
+Let's see MOLGENIS Compute in action!
+
+Go back to the molgenis compute directory and generate the scripts:
 
 ```bash
-  cd myfirst_workflow
-  ls
+sh molgenis_compute.sh --generate --parameters myfirst_workflow/parameters.csv --workflow myfirst_workflow/workflow.csv --defaults myfirst_workflow/workflow.defaults.csv
 ```
 
-The directory contains a typical Molgenis Compute workflow structure
+Take a look at the generated scripts in `rundir`
+
+file | description
+---------|----
+`step1_0.sh`, `step1_1.sh`|	 the two scripts for `step1`, one for each value of the `input` parameter
+`step2_0.sh`| the script for `step2`
+`submit.sh`| a submission script, which will run the generated script files in the correct order
+`user.env`| the user environment, containing runtime parameter values as input for `step1`
```

Comment:
I am unsure, I tossed it away and the whole thing ran just fine. I don't see it get mentioned anywhere in the scripts either.